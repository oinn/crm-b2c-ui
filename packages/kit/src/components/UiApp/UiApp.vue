<template>
  <div :class="classes">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import { themeable } from '@/mixins';

export default defineComponent({
  name: 'UiApp',
  mixins: [themeable],
  computed: {
    classes() {
      return {
        'ui-app': true,
        ...this.themeClasses,
      };
    },
  },
  created() {
    this.$ui.theme.isDark = this.dark || false;
  },
});
</script>

<style lang="scss">
// noinspection CssUnknownTarget
@import "@/styles/colors/index.scss";

.ui-app {
  height: 100%;
  display: flex;
  flex-direction: column;
  overflow: auto;
  position: relative;

  font-family: Rostelecom Basis, Arial, Helvetica Neue, Helvetica, sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;

  &.ui-theme--light {
    background-color: get-element-color("light", "background");

    color: get-element-color("light", "text-primary");
  }

  &.ui-theme--dark {
    background-color: get-element-color("dark", "background");

    color: get-element-color("dark", "text-primary");
  }

  * {
    box-sizing: border-box;
  }
}
</style>
