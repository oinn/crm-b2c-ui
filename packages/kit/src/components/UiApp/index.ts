export * as UiApp from './UiApp.vue';
