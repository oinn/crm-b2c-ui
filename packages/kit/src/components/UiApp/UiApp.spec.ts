import { mount } from '@vue/test-utils';
import UiApp from './UiApp.vue';

describe('UiApp', () => {
  it('Rendered correctly', () => {
    const wrapper = mount(UiApp, {
      slots: {
        default: 'Default slot text',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
