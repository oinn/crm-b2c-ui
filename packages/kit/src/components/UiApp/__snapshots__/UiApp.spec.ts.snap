// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiApp Rendered correctly 1`] = `"<div class=\\"ui-app\\">Default slot text</div>"`;
